django/contrib/admin/views/main.py

```diff
@@ -135,6 +138,32 @@ def get_filters(self, request):
             if spec and spec.has_output():
                 filter_specs.append(spec)
 
+        if self.date_hierarchy:
+            year = lookup_params.pop('{}__year'.format(self.date_hierarchy), None)
+            if year is not None:
+                month = lookup_params.pop('{}__month'.format(self.date_hierarchy), None)
+                day = lookup_params.pop('{}__day'.format(self.date_hierarchy), None)
+
+                from_date = datetime.datetime(
+                    int(year),
+                    int(month if month is not None else 1),
+                    int(day if day is not None else 1),
+                )
+                if settings.USE_TZ:
+                    from_date = make_aware(from_date)
+
+                if day:
+                    to_date = from_date + datetime.timedelta(days=1)
+
+                elif month:
+                    to_date = (from_date + datetime.timedelta(days=32)).replace(day=1)
+
+                else:
+                    to_date = from_date.replace(year=from_date.year + 1)
+
+                lookup_params['{}__gte'.format(self.date_hierarchy)] = from_date
+                lookup_params['{}__lt'.format(self.date_hierarchy)] = to_date
```

Comment:
I think we should protect this filter against invalid inputs from users, e.g.:
```
{}__year=a
{}__year=0
{}__month=a
{}__month=0
{}__month=13
{}__day=a
{}__day=0
{}__day=32
```
Currently it throws `ValueError` in most cases.

Reply:
We can catch ValueError when from_date is created (few lones above). Catching `ValueError` will cover both invalid values ('x,'y','z') and invalid dates (month 13 etc...).

Would `IncorrectLookupParameters` be appropriate?